0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,662 +116,889 @@ informative:
+The recommended modulus size as of 2016 is 2048 bits [KEYLENGTH-RECOMMENDATION](#KEYLENGTH-RECOMMENDATION). In the future we anticipate an upgrade to 3072 bits which provides approximately 128 bits of security [NIST-KEYMANAGEMENT](#NIST-KEYMANAGEMENT) (p. 64), about the same level as SHA-256.
 
+### Cost {#rsa-sha-256-condition-type-cost}
 
-### Fulfillment {#rsa-sha-256-condition-type-fulfillment}
+The cost is the square of the RSA key modulus size (in bits) divided by the constant 64.
 
-    RsaSha256FulfillmentPayload ::= SEQUENCE {
-      modulus OCTET STRING (SIZE(128..512)),
-      signature OCTET STRING (SIZE(128..512))
+    cost = ( (modulus size in bits) ^ 2 ) / 64 
+
+### ASN.1 {#rsa-sha-256-condition-asn1}
+
+    -- Condition Fingerprint
+    RsaFingerprintContents ::= SEQUENCE {
+      modulus              OCTET STRING
     }
 
+    -- Fulfillment 
+    RsaSha256Fulfillment ::= SEQUENCE {
+      modulus              OCTET STRING,
+      signature            OCTET STRING
+    }
+
+### Condition Format {#rsa-sha-256-condition-type-condition}
+The fingerprint of an RSA-SHA-256 condition is the SHA-256 digest of the DER encoded fingerprint contents which is a SEQUENCE of a single element, the modulus of the RSA Key Pair.
+
+### Fulfillment Format {#rsa-sha-256-condition-type-fulfillment}
+
+The fulfillment of an RSA-SHA-256 crypto-condition is an RsaSha256Fulfillment which is a SEQUENCE of:
 
 modulus
-: is an octet string representing the RSA public modulus in big-endian byte order. See [rsa-sha-256-condition-type-condition](#rsa-sha-256-condition-type-condition)  
+: The modulus of the RSA key pair used to sign and verify the signature provided. 
 
 signature
-: is an octet string representing the RSA signature. It MUST be encoded in big-endian byte order with the exact same number of octets as the modulus, even if this means adding leading zeros. This ensures that the fulfillment size is constant and known ahead of time. Note that the field is still binary encoded with a length prefix for consistency.
+: An octet string representing the RSA signature on the message M.
 
-: Implementations MUST verify that the signature and modulus consist of the same number of octets and that the signature is numerically less than the modulus.
+: Implementations MUST verify that the signature is numerically less than the modulus.
 
-The message to be signed is provided separately. If no message is provided, the message is assumed to be an octet string of length zero.
+Note that the message that has been signed is provided separately. If no message is provided, the message is assumed to be an octet string of length zero.
 
-### Implementation {#rsa-sha-256-condition-type-implementation}
-The recommended modulus size as of 2016 is
```

Comment:
I can't see where `ED25519-SHA-256` is depending on SHA-512. Am I missing something?